packages/lodestar/src/sync/sync.ts

```diff
@@ -166,14 +167,16 @@ export class BeaconSync implements IBeaconSync {
 
   private onUnknownBlockRoot = async (root: Root): Promise<void> => {
     const peerBalancer = new RoundRobinArray(this.getPeers());
-    let peer = peerBalancer.next();
-    let block: SignedBeaconBlock;
-    while (!block && peer) {
-      block = (await this.network.reqResp.beaconBlocksByRoot(peer, [root]))[0];
-      peer = peerBalancer.next();
-    }
-    if(block) {
-      await this.chain.receiveBlock(block);
+    const maxAttempts = 1000;
+    for (let attempt = 0; attempt < maxAttempts; attempt++) {
```

Comment:
why switch to for loop?

Reply:
@wemeetagain Commented he got stuck in an infinite loop sometimes in this function, so it might make sense to add some abort mechanism, i.e. a limited loop. No clue on what would be a reasonable maxAttempts value here, nor if it's the best approach to prevent the infinite loop possibility without addressing the underlying cause.

Reply:
oh yeah, because someone changed peerBalalncer to loop around. I changed that in regular sync v2.
So we have two options, either reset RoundRobinArray to return null if all peers are discarded or reduce maxAttempts here to this.getPeers().length x 2 (so we try each peer twice).

if you have 10 peers and block doesn't exists this will surely make you banned from those 10 peers